Packs/FireEyeEX/Integrations/FireEyeEX/FireEyeEX.py

```diff
@@ -0,0 +1,575 @@
+    client.fe_client.create_blockedlist_request(type_, entry_value, matches)
+
+    return CommandResults(
+        readable_output=f'Blockedlist entry {entry_value} of type {type_} was created.'
+    )
+
+
+@logger
+def update_blockedlist(client: Client, args: Dict[str, Any]) -> CommandResults:
+    type_ = args.get('type', '')
+    entry_value = args.get('entry_value', '')
+    matches = int(args.get('matches', '0'))
+
+    # check that the entry_value does exist
+    exist = False
+    current_allowed_list = client.fe_client.list_blockedlist_request(type_)
+    for entry in current_allowed_list:
+        if entry_value == entry.get('name'):
+            exist = True
+    if not exist:
+        raise DemistoException(str(f'Cannot update the entry_value {entry_value} as it does not exist in the '
+                                   f'Blockedlist of type {type_}.'))
+
+    # gets 200 back without content if successful
+    client.fe_client.update_blockedlist_request(type_, entry_value, matches)
+
+    return CommandResults(
+        readable_output=f'Blockedlist entry {entry_value} of type {type_} was updated.'
+    )
+
+
+@logger
+def delete_blockedlist(client: Client, args: Dict[str, Any]) -> CommandResults:
+    type_ = args.get('type', '')
+    entry_value = args.get('entry_value', '')
+
+    # check that the entry_value does exist
+    exist = False
+    current_allowed_list = client.fe_client.list_blockedlist_request(type_)
+    for entry in current_allowed_list:
+        if entry_value == entry.get('name'):
+            exist = True
+    if not exist:
+        raise DemistoException(str(f'Cannot delete the entry_value {entry_value} as it does not exist in the '
+                                   f'Blockedlist of type {type_}.'))
+
+    # gets 200 back without content if successful
+    client.fe_client.delete_blockedlist_request(type_, entry_value)
+
+    return CommandResults(
+        readable_output=f'Blockedlist entry {entry_value} of type {type_} was deleted.'
+    )
+
+
+@logger
+def fetch_incidents(client: Client, last_run: dict, first_fetch: str, max_fetch: int = 50,
+                    info_level: str = 'concise') -> Tuple[dict, list]:
+    if not last_run:  # if first time fetching
+        next_run = {
+            'time': to_fe_datetime_converter(first_fetch),
+            'last_alert_ids': []
+        }
+    else:
+        next_run = last_run
+    demisto.info(f'{INTEGRATION_NAME} executing fetch with: {str(next_run.get("time"))}')
+
+    raw_response = client.fe_client.get_alerts_request(request_params={
+        'start_time': to_fe_datetime_converter(first_fetch),
+        'info_level': info_level
+    })
+    all_alerts = raw_response.get('alert')
+
+    if not all_alerts:
+        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
```

Comment:
if we did not fetch any alerts, we can just use the time from the run before. using time from noe is creating a race new alerts that can be created now.

also, I think you should use next run, but not sure about `last_alert_ids`

Reply:
the problem is that the `duration` of the search can not be more than 48 hours(api limitation), which is the reason i'm advancing the time even if no alerts were fetched.
agrees with the 2nd point.